### PR TITLE
Store origin data alongside localized data

### PR DESCRIPTION
**Pull request by ryanmitchell**
As discussed [here](https://github.com/statamic/eloquent-driver/issues/112), we should store origin entry data alongside localisation data so it can be queried at the database level.

Let me know if youre happy with this approach as I'll need to add some test coverage on this before it should be merged.

### CODE REVIEW

**Comment by ryanmitchell** [comment]
@FrittenKeeZ yep, I was battling the tests a bit so moved this to draft. Should be working now with the latest commit.

File: src/Entries/Entry.php [hunk 1]
```diff
@@ -36,17 +38,39 @@ public function toModel()
...
+            }
+
+            if ($origin) {
+                $localizedFields = $blueprint
+                    ->fields()
+                    ->localizable()
+                    ->all()
+                    ->map
+                    ->handle()
+                    ->all();
```
[the remark this answers is not in the document]
Comment (ryanmitchell) [reply src/Entries/Entry.php:56]: So get the data for the entry for those fields and filter it to remove blank values?

**Comment by FrittenKeeZ** [comment]
@ryanmitchell @jasonvarga we're facing rather big performance issues with `blueprint` being stored in json instead of its own field, so I just wanted to hear you out if you're okay with pulling `blueprint` out of json with an index so queries don't take 5+ seconds to complete?

File: src/Entries/Entry.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanmitchell) [reply src/Entries/Entry.php]: I've gone ahead and [implemented this](https://github.com/statamic/eloquent-driver/pull/114/commits/352427fd2ef8a6a9ec73957327333b34eb55fdae) - let me know if you want a different approach

**Comment by FrittenKeeZ** [comment]
@ryanmitchell do you have any sense of whether this is ready for use? The missing `blueprint` on translated entries are causing us big issues currently, so I'm torn between using this patch to get the missing data or creating a PR for storing `blueprint` in a separate field in the database for improving performance as well.

**Comment by ryanmitchell** [comment]
As far as I can see it is, but Jason may feel differently

**Comment by FrittenKeeZ** [comment]
We just talked it through with the team and I'll be making a PR to fix #126 soon, as it will fix three issues we have regarding performance, code complexity and data duplication for blueprint.

File: src/Entries/Entry.php [hunk 2]
```diff
@@ -42,15 +44,48 @@ public function toModel()
 
         $data = $this->data();
 
+        $origin = $this->origin();
+        $date = $this->hasDate() ? $this->date() : null;
+
+        if ($blueprint = $this->blueprint()) {
```
Comment (FrittenKeeZ) [inline src/Entries/Entry.php:50]: @ryanmitchell We ran into a few issues with this block regarding fields not properly being detected as inherited, which is why I made a few changes to my PR.
https://github.com/statamic/eloquent-driver/pull/144/files#diff-31fcf8a95fd0dbc8da9e8dc7ae856aa51bd921c50a8059e23df7d411737ab65fR47

Comment (ryanmitchell) [reply src/Entries/Entry.php:50]: Can you explain the issues you were seeing?

Comment (FrittenKeeZ) [reply src/Entries/Entry.php]: Reverting through CP - I guess through code it would just be unsetting the fields?

Comment (ryanmitchell) [reply src/Entries/Entry.php]: Got you now, and was able to replicate it.
I've pushed a fix for that.

**Comment by FrittenKeeZ** [comment]
@ryanmitchell I can see I added some additional handling regarding `parent`, but I can't remember whether it was an issue or I just added it as a precaution.
https://github.com/statamic/eloquent-driver/pull/144/files#diff-31fcf8a95fd0dbc8da9e8dc7ae856aa51bd921c50a8059e23df7d411737ab65fR71

**Comment by ryanmitchell** [comment]
Yeah we're not storing parent going forward - its redundant as we have origin as a column.

**Comment by FrittenKeeZ** [comment]
> Ah right, so german might be the parent of french, but english is the fallback? Ok, storing the ID is fine (as your PR did) - I think until this PR its storing the whole entry which is a lot.

Not quite... each site / language is its own regarding parents, like so:
```
                 Origin
Contact (EN)  - - - - - - - Kontakt (DE)
    |                         |
    | Parent                  | Parent
    |            Origin       |
About US (EN) - - - - - - - Über Uns (DE)
```

**Comment by ryanmitchell** [comment]
@FrittenKeeZ are you happy that this covers everything you found now?

**Review by FrittenKeeZ — APPROVED** [review]

**Comment by ryanmitchell** [comment]
Thanks @FrittenKeeZ

**Pull request merged**